### PR TITLE
Add `sig` to the keywords list

### PR DESCRIPTION
**Pull request by auduchinok**
Fixes `sig` keyword isn't escaped by `PrettyNaming.NormalizeIdentifierBackticks`, which breaks parsing if such an identifier if produced by a feature in an IDE.

### CODE REVIEW

File: src/Compiler/SyntaxTree/PrettyNaming.fs [hunk 1]
```diff
@@ -224,6 +224,7 @@ let keywordsWithDescription: (string * string) list =
         "rec", FSComp.SR.keywordDescriptionRec ()
         "return", FSComp.SR.keywordDescriptionReturn ()
         "return!", FSComp.SR.keywordDescriptionReturnBang ()
+        "sig", FSComp.SR.keywordDescriptionSig ()
```
Comment (psfinaki) [inline src/Compiler/SyntaxTree/PrettyNaming.fs:227]: Are you sure this is the right place? There are more of them in the [keyword reference](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/keyword-reference) and all the others are a bit elsewhere in this file.

(and yeah I wonder how this particular one got slipped)

Comment (auduchinok) [reply src/Compiler/SyntaxTree/PrettyNaming.fs:227]: This is where the API that does the escaping look for the keywords.

**Review by psfinaki — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Review by 0101 — APPROVED** [review]

File: src/Compiler/SyntaxTree/PrettyNaming.fs (hunk 1 above)
Comment (auduchinok) [reply src/Compiler/SyntaxTree/PrettyNaming.fs:227]: > There are more of them in the [keyword reference](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/keyword-reference) and all the others are a bit elsewhere in this file.

Hmm, maybe we should do a systematic update of that list then

Comment (psfinaki) [reply src/Compiler/SyntaxTree/PrettyNaming.fs:227]: @KevinRansom @vzarytovskii any thoughts on this? I'm not very strong at this part here, but it feels inconsistent here now.

**Pull request merged**